constraints/miden-vm/hash.air

```diff
@@ -0,0 +1,205 @@
+mod HashChipletAir
+
+### Constants and periodic columns ################################################################
+
+periodic_columns:
+    cycle_row_0: [1, 0, 0, 0, 0, 0, 0, 0]
+    cycle_row_6: [0, 0, 0, 0, 0, 0, 1, 0]
+    cycle_row_7: [0, 0, 0, 0, 0, 0, 0, 1]
+
+
+### Helper functions ##############################################################################
+
+# Returns binary negation of the value.
+fn binary_not(value: scalar) -> scalar:
+    return 1 - value
+
+
+# Set to 1 when selector flags are (1,0,1) on rows which are multiples of 8. This is flag of 
+# the instruction that initiates Merkle path verification computation.
+fn get_f_mp(s: vector[3]) -> scalar:
+    return cycle_row_0 * s[0] * binary_not(s[1]) * s[2]
```

Comment:
nit: should we use logical operators here and other flag calculations as well?

Reply:
Yes, I think we should do that. Thank you!